sim/src/tlv.rs

```diff
@@ -569,13 +636,15 @@ impl ManifestGen for TlvGen {
             buf.append(&mut cipherkey);
 
             if self.kinds.contains(&TlvKinds::ENCEC256) {
-                assert!(buf.len() == 113);
+                let size = if aes256 { 129 } else { 113 };
+                assert!(buf.len() == 129 || buf.len() == 113);
```

Comment:
Could not this be `assert!(buf.len() == size)` since you are already know which size it's supposed to be?